capsules/src/process_console.rs

```diff
@@ -561,6 +630,22 @@ impl<'a, A: Alarm<'a>, C: ProcessManagementCapability> ProcessConsole<'a, A, C>
                     Ok(s) => {
                         let clean_str = s.trim();
 
+                        // Try to add a new command to the history buffer
+                        if clean_str.len() > 0 && COMMAND_HISTORY_LEN > 0 {
```

Comment:
Split this `if` in two, as the compiler will be able to reduce it. The constant has to be checked first.

Reply:
Thought it would be this way, however can you explain a bit how it will "reduce it"; is it a rustc optimization or general compiler design ? Just curious, nothing more.